### PR TITLE
Add support for REGEXP_REPLACE()

### PR DESCRIPTION
**Pull request by captain-igloo**

### CODE REVIEW

**Comment by captain-igloo** [comment]
I have updated the docs and added a version without flags.

File: docs/AVAILABLE-FUNCTIONS-AND-OPERATORS.md [hunk 1]
```diff
@@ -72,6 +72,8 @@
 | regexp_like (with no flags) | REGEXP_LIKE | `MartinGeorgiev\Doctrine\ORM\Query\AST\Functions\RegexpLike` |
 | regexp_match (with flags) | FLAGGED_REGEXP_MATCH | `MartinGeorgiev\Doctrine\ORM\Query\AST\Functions\FlaggedRegexpMatch` |
 | regexp_match (with no flags) | REGEXP_MATCH | `MartinGeorgiev\Doctrine\ORM\Query\AST\Functions\RegexpMatch` |
+| regexp_replace (with flags) | FLAGGED_REXEXP_REPLACE | `MartinGeorgiev\Doctrine\ORM\Query\AST\Functions\FlaggedRegexpReplace` |
```
Comment (martin-georgiev) [inline docs/AVAILABLE-FUNCTIONS-AND-OPERATORS.md:75]:
```suggestion
| regexp_replace (with flags) | FLAGGED_REGEXP_REPLACE | `MartinGeorgiev\Doctrine\ORM\Query\AST\Functions\FlaggedRegexpReplace` |
```

File: docs/AVAILABLE-FUNCTIONS-AND-OPERATORS.md [hunk 2]
```diff
@@ -72,6 +72,8 @@
...
+| regexp_replace (with no flags) | REXEXP_REPLACE | `MartinGeorgiev\Doctrine\ORM\Query\AST\Functions\RegexpReplace` |
```
Comment (martin-georgiev) [inline docs/AVAILABLE-FUNCTIONS-AND-OPERATORS.md:76]:
```suggestion
| regexp_replace (with no flags) | REGEXP_REPLACE | `MartinGeorgiev\Doctrine\ORM\Query\AST\Functions\RegexpReplace` |
```

File: src/MartinGeorgiev/Doctrine/ORM/Query/AST/Functions/FlaggedRegexpReplace.php [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+
+namespace MartinGeorgiev\Doctrine\ORM\Query\AST\Functions;
+
+/**
+ * Implementation of PostgreSql REGEXP_REPLACE().
+ *
+ * @see https://www.postgresql.org/docs/15/functions-matching.html#FUNCTIONS-POSIX-REGEXP
+ * @since 2.5.0
+ *
+ * @author Martin Georgiev <martin.georgiev@gmail.com>
```
Comment (martin-georgiev) [inline src/MartinGeorgiev/Doctrine/ORM/Query/AST/Functions/FlaggedRegexpReplace.php:13]:
```suggestion
 * @author Colin Doig
```

File: src/MartinGeorgiev/Doctrine/ORM/Query/AST/Functions/RegexpReplace.php [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+
+namespace MartinGeorgiev\Doctrine\ORM\Query\AST\Functions;
+
+/**
+ * Implementation of PostgreSql REGEXP_REPLACE().
+ *
+ * @see https://www.postgresql.org/docs/15/functions-matching.html#FUNCTIONS-POSIX-REGEXP
+ * @since 2.5.0
+ *
+ * @author Martin Georgiev <martin.georgiev@gmail.com>
```
Comment (martin-georgiev) [inline src/MartinGeorgiev/Doctrine/ORM/Query/AST/Functions/RegexpReplace.php:13]:
```suggestion
 * @author Colin Doig
```

**Review by martin-georgiev — APPROVED** [review]

**Comment by martin-georgiev** [comment]
Thank you @captain-igloo 😊

**Comment by captain-igloo** [comment]
thanks - can you release a new version which includes this?

**Comment by martin-georgiev** [comment]
Done - v2.5.0

**Pull request merged**